api/storage_service.cc

```diff
@@ -668,21 +668,14 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         auto keyspace = validate_keyspace(ctx, req->param);
         auto table_infos = parse_table_infos(keyspace, ctx, req->query_parameters, "cf");
         apilog.debug("force_keyspace_compaction: keyspace={} tables={}", keyspace, table_infos);
+
+        auto& cm = db.local().get_compaction_manager();
+        auto compaction_module_ptr = cm.get_compaction_module().shared_from_this();
+        auto task_impl_ptr = std::make_unique<major_keyspace_compaction_task_impl>(compaction_module_ptr, keyspace, db, table_infos);
+        auto task = co_await compaction_module_ptr->make_task(std::move(task_impl_ptr), {});
+        task->start();
         try {
-            co_await db.invoke_on_all([&] (replica::database& db) -> future<> {
-                auto local_tables = table_infos;
-                // major compact smaller tables first, to increase chances of success if low on space.
-                std::ranges::sort(local_tables, std::less<>(), [&] (const table_info& ti) {
-                    try {
-                        return db.find_column_family(ti.id).get_stats().live_disk_space_used;
-                    } catch (const replica::no_such_column_family& e) {
-                        return int64_t(-1);
-                    }
-                });
-                co_await run_on_existing_tables("force_keyspace_compaction", db, keyspace, local_tables, [] (replica::table& t) {
-                    return t.compact_all_sstables();
-                });
-            });
+            co_return task->done();
```

Comment:
should probably be `co_return co_await task->done()`

Reply:
It won't implicitly cast to `json_void`